include/boost/unordered/detail/archive_constructed.hpp

```diff
@@ -54,15 +54,15 @@ struct archive_constructed:private noncopyable
 #pragma GCC diagnostic ignored "-Wstrict-aliasing"
 #endif
 
-  T& get(){return *reinterpret_cast<T*>(&space);}
+  T& get(){return *space.address();}
 
 #if defined(BOOST_UNORDERED_IGNORE_WSTRICT_ALIASING)
 #pragma GCC diagnostic pop
 #undef BOOST_UNORDERED_IGNORE_WSTRICT_ALIASING
 #endif
 
 private:
-  typename aligned_storage<sizeof(T),alignment_of<T>::value>::type space;
+  opt_storage<T> space;
```

Comment:
It does the same thing, just abstracted away so that we can attach methods to it.

In addition, it's automatically appropriately sized and aligned for our purposes. I got the trick from Peter, actually, and I've just chosen to stick with it.

Reply:
Ok!